### ISSUE TITLE
All docker containers/tpot fail, do not come back after a reboot

### ISSUE THREAD

**Issue by amishrabbit** [dtag-dev-sec/tpotce#511]
I hadn't been paying attention to my tpot for a couple of weeks and when I checked in on it on Sunday, it was just idle with no services running. I wiped the VM hard drive and reinstalled from scratch using the 19.03.1 ISO. On the first try at reinstalling, nothing worked right, so I wiped it and tried again, and it worked great...for about 36 hours.

About 30 minutes ago, I noticed that I suddenly couldn't get to Kibana, so I checked the container status in the :64294 GUI and it reports all containers are down. 
![system monitor](https://user-images.githubusercontent.com/2653726/71950594-11fe7b80-3195-11ea-88f3-abb537c3ff67.png)

I rebooted the box and it's been idling and not doing anything since reboot. Manually restarting the containers one by one fails each time; They try to start up and immediately quit. The tpot service is dead; docker network, volume, etc report FAILURE

The system was under sustained probing for the entire 36 hours it was online. 

I cannot determine why this keeps happening, but this is now the fourth time in the past six months that I have discovered a similar situation, tried to perform a full reinstall, and then had the entire system go down within a matter of days and not come back.

Happy to work with you to try to figure this out.

## ?? Basic support information (commands are expected to run as `root`)

- What version of the OS are you currently using `lsb_release -a` and `uname -a`?

No LSB modules are available.
Distributor ID: Debian
Description:    Debian GNU/Linux bullseye/sid
Release:        unstable
Codename:       sid

Linux xxxxxxxxxxxxxxxxxx 5.4.0-1-amd64 #1 SMP Debian 5.4.6-1 (2019-12-27) x86_64 GN       

- What T-Pot version are you currently using?

19.03.1

- What edition (Standard, Nextgen, etc.) of T-Pot are you running?

Nextgen

- What architecture are you running on (i.e. hardware, cloud, VM, etc.)?

VMWare vSphere VM

- Did you have any problems during the install? If yes, please attach `/install.log` `/install.err`.

no

- How long has your installation been running?

36 hours

- Did you install upgrades, packages or use the update script?

no

- Did you modify any scripts or configs? If yes, please attach the changes.

n/a 

- Please provide a screenshot of `glances` and `htop`.
![glances](https://user-images.githubusercontent.com/2653726/71950376-30b04280-3194-11ea-9640-18031bc5568f.png)
![htop](https://user-images.githubusercontent.com/2653726/71950384-34dc6000-3194-11ea-8fd9-037f6abff356.png)

- How much free disk space is available (`df -h`)?

Filesystem      Size  Used Avail Use% Mounted on
udev             12G     0   12G   0% /dev
tmpfs           2.4G   56M  2.3G   3% /run
/dev/sda2       496G   17G  454G   4% /
tmpfs            12G     0   12G   0% /dev/shm
tmpfs           5.0M     0  5.0M   0% /run/lock
tmpfs            12G     0   12G   0% /sys/fs/cgroup
tmpfs           2.4G     0  2.4G   0% /run/user/1000

- What is the current container status (`dps.sh`)?

No containers are running. The output of this shell command is as follows: 
./dps.sh: 20: function: not found
./dps.sh: 22: Syntax error: "}" unexpected

- What is the status of the T-Pot service (`systemctl status tpot`)?
![tpot status](https://user-images.githubusercontent.com/2653726/71950317-fb0b5980-3193-11ea-849b-ff6a338c54a5.png)
![status tpot](https://user-images.githubusercontent.com/2653726/71950337-10808380-3194-11ea-8e06-fb08538fa793.png)

● tpot.service - tpot
     Loaded: loaded (/etc/systemd/system/tpot.service; enabled; vendor preset: >
     Active: activating (start-pre) since Wed 2020-01-08 03:41:53 UTC; 13min ago
    Process: 933 ExecStartPre=/opt/tpot/bin/updateip.sh (code=exited, status=0/>
Cntrl PID: 964 (clean.sh)
      Tasks: 13 (limit: 28832)
     Memory: 1.7G
     CGroup: /system.slice/tpot.service
             ├─ 964 /bin/bash /opt/tpot/bin/clean.sh on
             ├─1044 tar -I /usr/bin/pigz -cvf /data/dionaea/bistreams.tgz /data>
             ├─1045 /bin/sh -c /usr/bin/pigz
             └─1046 /usr/bin/pigz

- What ports are being occupied? Stop T-Pot `systemctl stop tpot` and run `netstat -tulpen`
![netstat](https://user-images.githubusercontent.com/2653726/71950352-1c6c4580-3194-11ea-8225-a9299bd572d5.png)

Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       User       Inode      PID/Program name
tcp        0      0 127.0.0.1:37789         0.0.0.0:*               LISTEN      0          18007      1092/cockpit-bridge
tcp        0      0 127.0.0.1:45757         0.0.0.0:*               LISTEN      1000       24036      1078/cockpit-bridge
tcp        0      0 0.0.0.0:64295           0.0.0.0:*               LISTEN      0          19491      681/sshd
tcp        0      0 0.0.0.0:61209           0.0.0.0:*               LISTEN      0          463        667/python3
tcp6       0      0 :::64294                :::*                    LISTEN      0          13752      1/init
tcp6       0      0 :::64295                :::*                    LISTEN      0          19493      681/sshd
udp        0      0 0.0.0.0:68              0.0.0.0:*                           0          2930       596/dhclient
udp        0      0 192.168.0.21:123        0.0.0.0:*                           0          18777      679/ntpd
udp        0      0 127.0.0.1:123           0.0.0.0:*                           0          18775      679/ntpd
udp        0      0 0.0.0.0:123             0.0.0.0:*                           0          18771      679/ntpd
udp6       0      0 :::123                  :::*                                0          18768      679/ntpd

- If a single container shows as `DOWN` you can run `docker logs <container-name>` for the latest log entries

They're all down

**Comment by jrsanchezsilva** [comment]
Good morning, I just saw the kibana and I have the same problem. So far I have solved it with this, the only reason I could think of that it is not working for you is that the automatic rules exception script does not work properly (/opt/tpot/bin/rules.sh). Since the script relies on yq which is installed via pip maybe its path needs rehashing.
Can you try (as root) and please only in this order:

systemctl stop tpot
hash -r
pip3 install yq
If you get feedback from yq you can start T-Pot again (systemctl start tpot), otherwise please post output of...

iptables-legacy -t raw -vL
... once you started all T-Pot containers.

Please check if it was fixed @amishrabbit 

On the contrary it would be to evaluate the response to what is really happening, please @t3chn0m4g3  your support for this case, in advance thank you very much.

**Comment by jrsanchezsilva** [comment]
Good morning, I still have the problem, I really don't know why the containers are down, please support @t3chn0m4g3, attached images.

![image](https://user-images.githubusercontent.com/51901933/72079411-df559f80-32c8-11ea-9f88-868aafd1b5b5.png)

![image](https://user-images.githubusercontent.com/51901933/72079553-2348a480-32c9-11ea-858f-1c130cdfb2e5.png)

**Comment by t3chn0m4g3** [comment]
Are grc and yq installed? Please show the lines 20-22 of dps.sh. Also check the paths with which.

**Comment by jrsanchezsilva** [comment]
Good afternoon. I enclose line 20-22 of the dsp.sh file. As well as, the confirmation of installation of grc e yq. The problem of kibana still persists.

![image](https://user-images.githubusercontent.com/51901933/72277674-560fd700-3600-11ea-9574-dbdce914accd.png)

![image](https://user-images.githubusercontent.com/51901933/72277837-b0a93300-3600-11ea-8e87-2befe62ebb9b.png)

**Comment by t3chn0m4g3** [comment]
Please run `update.sh` from `/opt/tpot`. This will fix `dps.sh`.
As for Kibana, I cannot reproduce, need more info from your side.

**Comment by t3chn0m4g3** [comment]
Can you do a fresh install in either VirtualBox oder VMWare and check if the problem persists?

**Comment by sengkoil** [comment]
> I hadn't been paying attention to my tpot for a couple of weeks and when I checked in on it on Sunday, it was just idle with no services running. I wiped the VM hard drive and reinstalled from scratch using the 19.03.1 ISO. On the first try at reinstalling, nothing worked right, so I wiped it and tried again, and it worked great...for about 36 hours.
> 
> About 30 minutes ago, I noticed that I suddenly couldn't get to Kibana, so I checked the container status in the :64294 GUI and it reports all containers are down.
> ![system monitor](https://user-images.githubusercontent.com/2653726/71950594-11fe7b80-3195-11ea-88f3-abb537c3ff67.png)
> 
> I rebooted the box and it's been idling and not doing anything since reboot. Manually restarting the containers one by one fails each time; They try to start up and immediately quit. The tpot service is dead; docker network, volume, etc report FAILURE
> 
> The system was under sustained probing for the entire 36 hours it was online.
> 
> I cannot determine why this keeps happening, but this is now the fourth time in the past six months that I have discovered a similar situation, tried to perform a full reinstall, and then had the entire system go down within a matter of days and not come back.
> 
> Happy to work with you to try to figure this out.
> 
> ## ?? Basic support information (commands are expected to run as `root`)
> * What version of the OS are you currently using `lsb_release -a` and `uname -a`?
> 
> No LSB modules are available.
> Distributor ID: Debian
> Description: Debian GNU/Linux bullseye/sid
> Release: unstable
> Codename: sid
> 
> Linux xxxxxxxxxxxxxxxxxx 5.4.0-1-amd64 #1 SMP Debian 5.4.6-1 (2019-12-27) x86_64 GN
> 
> * What T-Pot version are you currently using?
> 
> 19.03.1
> 
> * What edition (Standard, Nextgen, etc.) of T-Pot are you running?
> 
> Nextgen
> 
> * What architecture are you running on (i.e. hardware, cloud, VM, etc.)?
> 
> VMWare vSphere VM
> 
> * Did you have any problems during the install? If yes, please attach `/install.log` `/install.err`.
> 
> no
> 
> * How long has your installation been running?
> 
> 36 hours
> 
> * Did you install upgrades, packages or use the update script?
> 
> no
> 
> * Did you modify any scripts or configs? If yes, please attach the changes.
> 
> n/a
> 
> * Please provide a screenshot of `glances` and `htop`.
>   ![glances](https://user-images.githubusercontent.com/2653726/71950376-30b04280-3194-11ea-9640-18031bc5568f.png)
>   ![htop](https://user-images.githubusercontent.com/2653726/71950384-34dc6000-3194-11ea-8fd9-037f6abff356.png)
> * How much free disk space is available (`df -h`)?
> 
> Filesystem Size Used Avail Use% Mounted on
> udev 12G 0 12G 0% /dev
> tmpfs 2.4G 56M 2.3G 3% /run
> /dev/sda2 496G 17G 454G 4% /
> tmpfs 12G 0 12G 0% /dev/shm
> tmpfs 5.0M 0 5.0M 0% /run/lock
> tmpfs 12G 0 12G 0% /sys/fs/cgroup
> tmpfs 2.4G 0 2.4G 0% /run/user/1000
> 
> * What is the current container status (`dps.sh`)?
> 
> No containers are running. The output of this shell command is as follows:
> ./dps.sh: 20: function: not found
> ./dps.sh: 22: Syntax error: "}" unexpected
> 
> * What is the status of the T-Pot service (`systemctl status tpot`)?
>   ![tpot status](https://user-images.githubusercontent.com/2653726/71950317-fb0b5980-3193-11ea-849b-ff6a338c54a5.png)
>   ![status tpot](https://user-images.githubusercontent.com/2653726/71950337-10808380-3194-11ea-8e06-fb08538fa793.png)
> 
> ● tpot.service - tpot
> Loaded: loaded (/etc/systemd/system/tpot.service; enabled; vendor preset: >
> Active: activating (start-pre) since Wed 2020-01-08 03:41:53 UTC; 13min ago
> Process: 933 ExecStartPre=/opt/tpot/bin/updateip.sh (code=exited, status=0/>
> Cntrl PID: 964 (clean.sh)
> Tasks: 13 (limit: 28832)
> Memory: 1.7G
> CGroup: /system.slice/tpot.service
> ├─ 964 /bin/bash /opt/tpot/bin/clean.sh on
> ├─1044 tar -I /usr/bin/pigz -cvf /data/dionaea/bistreams.tgz /data>
> ├─1045 /bin/sh -c /usr/bin/pigz
> └─1046 /usr/bin/pigz
> 
> * What ports are being occupied? Stop T-Pot `systemctl stop tpot` and run `netstat -tulpen`
>   ![netstat](https://user-images.githubusercontent.com/2653726/71950352-1c6c4580-3194-11ea-8225-a9299bd572d5.png)
> 
> Active Internet connections (only servers)
> Proto Recv-Q Send-Q Local Address Foreign Address State User Inode PID/Program name
> tcp 0 0 127.0.0.1:37789 0.0.0.0:* LISTEN 0 18007 1092/cockpit-bridge
> tcp 0 0 127.0.0.1:45757 0.0.0.0:* LISTEN 1000 24036 1078/cockpit-bridge
> tcp 0 0 0.0.0.0:64295 0.0.0.0:* LISTEN 0 19491 681/sshd
> tcp 0 0 0.0.0.0:61209 0.0.0.0:* LISTEN 0 463 667/python3
> tcp6 0 0 :::64294 :::* LISTEN 0 13752 1/init
> tcp6 0 0 :::64295 :::* LISTEN 0 19493 681/sshd
> udp 0 0 0.0.0.0:68 0.0.0.0:* 0 2930 596/dhclient
> udp 0 0 192.168.0.21:123 0.0.0.0:* 0 18777 679/ntpd
> udp 0 0 127.0.0.1:123 0.0.0.0:* 0 18775 679/ntpd
> udp 0 0 0.0.0.0:123 0.0.0.0:* 0 18771 679/ntpd
> udp6 0 0 :::123 :::* 0 18768 679/ntpd
> 
> * If a single container shows as `DOWN` you can run `docker logs <container-name>` for the latest log entries
> 
> They're all down

have occur the same problem with you..
since this case already open..i will keep following this thread and will update if any of the solution appear to be working.

**Comment by jrsanchezsilva** [comment]
Good afternoon, yesterday run update.sh is root mode (sudo su -), the update was done and then restart the server (physical server), so far all the containers and KIBANA are active.

image

image

Here we can realize that it was up 20 hours ago, hopefully it will continue that way since more than 36 hours or the next restart could not reach Kibana.

Why does the server restart automatically, what could be done in that case?
image
image

Despite the updates, the image I show shows available updates; what to do about it, just update ??
image

I will continue monitoring these days and I will tell you; Please support your questions in advance, thank you very much.

**Comment by t3chn0m4g3** [comment]
Thanks for the feedback. Will close this for now.

**Comment by amishrabbit** [comment]
Just wanted to circle back on this. I came down with a suspected case of Coronavirus a few days after filing this ticket and was sick for nearly a month, so I was not able to respond, and am just getting back to things I had left unfinished, like this ticket.

I have just deleted the previous VM, but I did discover before doing so that the sheer number of "bistreams" files in the dionaea storage area (/data/dionaea/bistreams) was causing the filesystem to choke. When I would delete them all, the problem resolved, but after a few days, the volume of inbound requests to hit dionaea would fill up the same area and the problem would repeat. 

I will try installing a NextGen (20.06) build and will report back if the problem repeats under that environment.
